code/game/objects/items/toys.dm

```diff
@@ -851,3 +839,24 @@
 	throwforce = 1
 	drop_sound = 'sound/items/drop/cardboardbox.ogg'
 	pickup_sound = 'sound/items/pickup/cardboardbox.ogg'
+
+/obj/item/toy/partypopper
+	name = "party popper"
+	desc = "Instructions : Aim away from face. Wait for appropriate timing. Pull cord, enjoy confetti."
+	icon_state = "partypopper"
+	w_class = ITEMSIZE_TINY
+	drop_sound = 'sound/items/drop/cardboardbox.ogg'
+	pickup_sound = 'sound/items/pickup/cardboardbox.ogg'
+
+/obj/item/toy/partypopper/attack_self(mob/user)
+	if(icon_state == "partypopper")
+		spark(src, 1, user.dir)
+		visible_message(SPAN_NOTICE("You pull on the string, releasing a burst of confetti!"), SPAN_NOTICE("[user] pulls on the string, releasing a burst of confetti!"))
```

Comment:
this order of operations is wrong, and it isn't called on user, so the self_message doesn't work